pkg/block/block.go

```diff
@@ -402,3 +402,22 @@ func MarkForNoCompact(ctx context.Context, logger log.Logger, bkt objstore.Bucke
 	level.Info(logger).Log("msg", "block has been marked for no compaction", "block", id)
 	return nil
 }
+
+// UnmarkBlock removes the file which marked the block for deletion or no-compact.
+func UnmarkBlock(ctx context.Context, logger log.Logger, bkt objstore.Bucket, id ulid.ULID, unmarkedBlock prometheus.Counter, markedFilename string) error {
+	markedFile := path.Join(id.String(), markedFilename)
+	markedFileExists, err := bkt.Exists(ctx, markedFile)
+	if err != nil {
+		return errors.Wrapf(err, "check if %s file exists in bucket", markedFile)
+	}
+	if !markedFileExists {
+		level.Warn(logger).Log("msg", "requested to unmark, but file does not exist; this should not happen; investigate", "err", errors.Errorf("file %s does not exist in bucket", markedFile))
```

Comment:
> "requested to unmark, but file does not exist; this should not happen; investigate"

This message is very explicit. Any particular reason to word it like this? Is it how other warnings in Thanos are worded?

Reply:
Yes, I tried to keep it similar to how it is written in other places. Can be rewritten, if required.